lib/auth/middleware.go

```diff
@@ -255,6 +256,14 @@ func (t *TLSServer) GetConfigForClient(info *tls.ClientHelloInfo) (*tls.Config,
 				return nil, trace.AccessDenied("access is denied")
 			}
 		}
+
+		// If Auth Service was dialed by SNI ALPN proxy (ProtocolAuth value present in SupportedProtos filed)
+		// reset cluster name in order to  validate against all known CAs.
+		// This is needed because in trusted cluster scenario the client uses root cluster cert with remote cluster name
+		// encoded into SNI value.
+		if utils.SliceContainsStr(info.SupportedProtos, alpnproxy.ProtocolAuth) {
```

Comment:
alternative approach is to encode cluster name value into SupportedProtos like `teleport-auth@enc_cluster_name_hex_val.teleport.cluster.local` and route based on `teleport-auth@` suffix.